Back-end/src/main/java/com/bramgussekloo/projects/controller/BuildingInstituteController.java

```diff
@@ -2,102 +2,77 @@
-        bi.getFromDatabase(id);
-        return new ResponseEntity<>(bi, HttpStatus.OK);
+    public BuildingInstituteController(BuildingInstituteService buildingInstituteService) {
+        this.buildingInstituteService = buildingInstituteService;
     }
 
     // Create a new BuildingInstitute object
-    @ApiOperation(value = "Create a new buildingInstitute")
+    @ApiOperation(value = "Link building to institute")
     @ApiResponses(value = {
-            @ApiResponse(code = 201, message = "Successfully created new buildingInstitute"),
+            @ApiResponse(code = 204, message = "Successfully created new buildingInstitute"),
             @ApiResponse(code = 400, message = "Bad request", response = Error.class),
             @ApiResponse(code = 401, message = "Bad credentials", response = Error.class),
             @ApiResponse(code = 500, message = "Internal Server Error", response = Error.class)
     })
     @PostMapping("admin/buildinginstitute")
-    @ResponseStatus(HttpStatus.CREATED)
-    private ResponseEntity<BuildingInstitute> createBuildingInstitute(
-            @ApiParam(value = "buildingInstitute that you want to add", required = true) @RequestBody BuildingInstitute buildingInstitute
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    private ResponseEntity<Void> createBuildingInstitute(
+            @ApiParam(value = "buildingInstitute that you want to add", required = true) @RequestBody Integer buildingId, Integer instituteId
     ) throws Exception {
-        buildingInstitute.createInDatabase();
-        return new ResponseEntity<>(buildingInstitute, HttpStatus.CREATED);
+        buildingInstituteService.linkBuildingInstitute(buildingId, instituteId);
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
     }
 
     // Delete a certain buildingInstitute object
     @ApiOperation(value = "Delete a certain buildingInstitute by id")
     @ApiResponses(value = {
-            @ApiResponse(code = 204, message = "Successfully deleted the buildingInstitute"),
+            @ApiResponse(code = 200, message = "Successfully deleted the buildingInstitute"),
             @ApiResponse(code = 400, message = "Bad request", response = Error.class),
             @ApiResponse(code = 401, message = "Bad credentials", response = Error.class),
             @ApiResponse(code = 500, message = "Internal Server Error", response = Error.class)
     })
-    @DeleteMapping("admin/buildinginstitute/{id}")
-    private ResponseEntity<Void> DeleteBuildingInstitute(
-            @ApiParam(value = "Id of the buildingInstitute that you want to delete", required = true) @PathVariable Integer id
-    ) throws Exception {
-        BuildingInstitute bi = new BuildingInstitute(id);
-        bi.deleteInDatabase();
-        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    @DeleteMapping("admin/buildinginstitute/{buildingId}")
+    @ResponseStatus(HttpStatus.OK)
+    private ResponseEntity<Set<Institute>> DeleteBuildingFromInstitutes(
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.